test/functional/p2p_orphan_handling.py

```diff
@@ -566,6 +569,47 @@ def test_orphan_txid_inv(self):
         assert tx_child["txid"] in node_mempool
         assert_equal(node.getmempoolentry(tx_child["txid"])["wtxid"], tx_child["wtxid"])
 
+    @cleanup
+    def test_max_orphan_amount(self):
+        self.log.info("Check that we never exceed our storage limits for orphans")
+
+        self.generate(self.wallet, 160)
+        node = self.nodes[0]
+        peer_1 = node.add_p2p_connection(P2PInterface())
+
+        self.log.info("Check that orphanage is empty on start of test")
+        self.wait_until(lambda: len(node.getorphantxs()) == 0)
```

Comment:
nit: why is this a `wait_until` instead of an assert?

Reply:
Thanks! Updated in [5c299ec](https://github.com/bitcoin/bitcoin/pull/31040/commits/5c299ecafe6f336cffa145d28036b04b87e26712)